src/7bb60_len_41b0.c

```diff
@@ -305,21 +848,191 @@ s32 phys_is_on_sloped_ground(void) {
     return ret;
 }
 
-INCLUDE_ASM(s32, "7bb60_len_41b0", phys_main_collision_below);
+void phys_main_collision_below(void) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    f32 collHeightHalf = playerStatus->colliderHeight * 0.5f;
+    f32 playerX = playerStatus->position.x;
+    f32 playerY = playerStatus->position.y + collHeightHalf;
+    f32 playerZ = playerStatus->position.z;
+    f32 outLength = playerStatus->colliderHeight;
+    f32 temp_f24 = (2.0f * playerStatus->colliderHeight) / 7.0f;
+    f32 hitRx, hitRz;
+    f32 hitDirX, hitDirZ;
+    s32 result;
+    s32 cond;
+
+    result = player_raycast_below_cam_relative(playerStatus, &playerX, &playerY, &playerZ, &outLength, &hitRx, &hitRz,
+                                               &hitDirX, &hitDirZ);
+    playerStatus->groundNormalPitch = get_player_normal_pitch();
+
+    if (collHeightHalf + (temp_f24 * 0.5f) < outLength) {
+        result = -1;
+    }
+    if (playerStatus->decorationList == 0) {
+        collisionStatus->currentFloor = result;
+    }
+    if (result >= 0) {
+        playerStatus->groundAnglesXZ.x = hitDirX;
+        playerStatus->groundAnglesXZ.y = hitDirZ;
+    }
+
+    if (func_800E0208()) {
+        return;
+    }
+
+    if (playerStatus->flags & (PLAYER_STATUS_FLAGS_FLYING | PLAYER_STATUS_FLAGS_FALLING | PLAYER_STATUS_FLAGS_JUMPING)) {
+        return;
+    }
+
+    if (playerStatus->actionState == ACTION_STATE_SLIDING) {
+        return;
+    }
+
+    if ((!(playerStatus->flags & PLAYER_STATUS_FLAGS_10) ||
+        (phys_adjust_cam_on_landing(), !phys_should_player_be_sliding()) ||
+        (set_action_state(ACTION_STATE_SLIDING), (playerStatus->actionState != ACTION_STATE_SLIDING))))
```

Comment:
This is incredibly cursed, perhaps this can be refactored (in the future) to not use the comma operator?